src/components/Exchange/FeesTooltip.tsx

```diff
@@ -0,0 +1,85 @@
+import { t, Trans } from "@lingui/macro";
+import ExternalLink from "components/ExternalLink/ExternalLink";
+import StatsTooltipRow from "components/StatsTooltip/StatsTooltipRow";
+import Tooltip from "components/Tooltip/Tooltip";
+import { getConstant } from "config/chains";
+import { BigNumberish } from "ethers";
+import { useChainId } from "lib/chains";
+import { USD_DECIMALS } from "lib/legacy";
+import { formatAmount, formatAmountFree } from "lib/numbers";
+
+type Row = {
+  label: string;
+  value: string;
+};
+
+function getExecutionFeeStr(chainId, executionFee, executionFeeUsd) {
+  const nativeTokenSymbol = getConstant(chainId, "nativeTokenSymbol");
+
+  if (!nativeTokenSymbol || !executionFee || !executionFeeUsd) {
+    return "...";
+  }
+
+  const formattedExecutionFee = formatAmountFree(executionFee, 18, 4);
+  const formattedExecutionFeeUsd = formatAmount(executionFeeUsd, USD_DECIMALS, 2);
+  return `${formattedExecutionFee} ${nativeTokenSymbol} ($${formattedExecutionFeeUsd})`;
+}
+
+type Props = {
+  totalFees: string;
+  fundingFee: string;
+  positionFee: string;
+  depositFee: string;
+  swapFee: string;
+  executionFees: { fee: BigNumberish; feeUSD: BigNumberish };
+  isOpening?: boolean;
+};
+
+function FeesTooltip({
+  totalFees,
+  fundingFee,
+  positionFee,
+  swapFee,
+  executionFees,
+  depositFee,
+  isOpening = true,
+}: Props) {
+  const { chainId } = useChainId();
+  const SWAP_FEE_LABEL = t`Swap Fee`;
+  const BORROW_FEE_LABEL = t`Borrow Fee`;
+  const executionFee = executionFees?.fee;
+  const executionFeeUSD = executionFees?.feeUSD;
+
+  const feeRows: Row[] = [
```

Comment:
it's a bit confusing with a lot of conditions on isOpening
maybe just:

```
const openingFees = isOpening ? [...] : []
const defaultFees = [...]

const feesRows = [
   ...openingFees
]
```

or as you did before but less conditions:
```
let feesRows: Row[] = [];

if(isOpening) {
 feesRows.push(
   {},
   {},
   {},
)
} else {
 feesRows.push(
   {},
   {},
   {},
)
}

feeRows.push(
  {},
  {}
)

feesRows = feesRows.filter(...)
```

Reply:
Done!